### PR TITLE
Refactor track scores

### LINKED ISSUE #15: If accessing a specific analysis state by using the URL, added score columns do not get re-computed
To reproduce:
1. Open list of all genes
2. Add a score column (e.g., expression of cell line `Becker`)
3. Copy URL and paste it in another browser tab

-> The score column gets added, however, no values are computed

BTW: the same problem occurs if one opens the provenance graph in an analysis, jumps to an early state (where a certain score column was not existing yet) and jumps back afterwards (the score column gets re-added, but no values are computed).

### PR DESCRIPTION
**Pull request by sgratzl**
closes #15

### CODE REVIEW

File: src/lineup/cmds.ts [hunk 1]
```diff
@@ -100,14 +107,14 @@ export function setColumn(provider:IObjectRef<any>, rid:number, path:string, pro
   });
 }
 
-async function addColumnImpl(inputs:IObjectRef<any>[], parameter:any) {
-  const p = await Promise.resolve((await inputs[0].v).data);
-  let ranking = p.getRankings()[parameter.rid];
+async function addColumnImpl(inputs:IObjectRef<IViewProvider>[], parameter:any) {
+  const p: ADataProvider = await Promise.resolve((await inputs[0].v).data);
```
Comment (chrilehner) [inline src/lineup/cmds.ts]: are the 2 awaits and the Promise.resolve wrapper necessary?

Comment (sgratzl) [inline src/lineup/cmds.ts]: `inputs[0].v` returns a type `Promise<{ data: Promise<ADataProvider>|ADataProvider>}>`

File: src/lineup/ALineUpView2.ts [hunk 2]
```diff
@@ -78,6 +83,7 @@ export abstract class ALineUpView2 extends AView {
 
     this.$node.classed('lineup', true);
 
+    // hack in for providing the data provider within the graph
```
Comment (sgratzl) [inline src/lineup/ALineUpView2.ts]: can you tell me one?

Comment (chrilehner) [inline src/lineup/ALineUpView2.ts]: no, I was just wondering why the resolve function is saved and then resolved below. I thought alternatively the `updateRef` function could return a promise and then resolve it or something. but I don't know most of that code, so I was just curious

Comment (sgratzl) [inline src/lineup/ALineUpView2.ts]: It is all about the reference. Using this method, the reference to `.data` remain the same, while using `Promise.resolve(...)` the value would change

Comment (chrilehner) [inline src/lineup/ALineUpView2.ts]: thank you for the explanation of this trick 👍

File: src/lineup/cmds.ts (hunk 1 above)
Comment (chrilehner) [inline src/lineup/cmds.ts]: ok, I see

**Pull request merged**